### ISSUE TITLE
Colour usage in code is not consistent.

### ISSUE THREAD

**Issue by eviltofu** [AlphaWallet/alpha-wallet-ios#3492]
How is colour defined in the app? I see colours defined in `Assests.xcassets` and `AppStyle.swift`. Both methods are used throughout the project. Can we stick with one? And put that in a readme doc?

**Comment by hboon** [comment]
@eviltofu This should evolve when, or in the preparation of building #3483. Most hex/rgb values will probably be defined in the assets catalog (at least to support Dark mode). We will probably still need some in `AppStyle.swift` though, eg. switching colors programmatically or giving them a semantic name.

Do you have a better suggestion to reduce it to just be in one place?

**Comment by eviltofu** [comment]
@hboon We should place all colours in `.xcasset` files. The colours can be given semantic style names. For `UIView` background colour, we can specify "UIView-Background" as the name and the light and dark colours, all in one entry. For other usages, we can put them into a `.xcassets` file. 

We should move all colours to `Colours.xcasset`. R.swift does pick up extra `.xcassets` files. I removed the `UIUserInterfaceStyle` entry in `info.plist` then created `Colour.xcasset` with a new colour, tofu (light-red, dark-blue), and set the background of the `Add Custom RPC Network` screen to tofu via `R.color.tofu()`. 

When in light mode, it renders the background in red and in dark mode blue. It does not change instantly when the style changes. I think it's due to some other setting someplace or perhaps the app was targeted to iOS 11.0

I created a new app targeted to iOS 15 with a simple coloured square. The square changes from red to blue as soon as I toggle the style change.
![Simulator Screen Shot - iPhone 13 - 2021-12-03 at 10 29 29](https://user-images.githubusercontent.com/1050309/144534881-60b839fa-b8cc-4507-8e01-3a8548c09756.png)![Simulator Screen Shot - iPhone 13 - 2021-12-03 at 10 30 28](https://user-images.githubusercontent.com/1050309/144534892-48e0017a-97a3-48b5-b37e-0c707e3ca6bd.png)

**Comment by hboon** [comment]
Yes, hex/rgb colors has to be defined in asset catalogs to better support Dark mode.

But I mean, we still need at least some to be defined in `AppStyle.swift` for when we need to explicitly switch colors dynamically. Another thing is the way the UI design system is set up, colors are given names (like `azure` and yes we do use `R.swift` for it, eg. `R.color.azure()`). We have to make the semantic ones like `UIView-Background` (your example) to one of those named colors. We can't do that mapping in the asset catalog (or, can we?).

**Comment by eviltofu** [comment]
Suggest we use a class to build the UI elements. 
```
class UiKitFactory: NSObject {
    ...
    func textField() -> UITextField {
        let textField = UITextField()
        textField.translateAutoresizingMask = false
        textField.backgroundColor = R.color.dove()
        ...
        return textField
    }
    ...
}
```

**Comment by hboon** [comment]
Doesn't rely on inheritance, easy to copy and paste and not so brittle. Might work?
